modules/dnn/src/op_inf_engine.cpp

```diff
@@ -545,6 +545,18 @@ void InfEngineBackendNet::initPlugin(InferenceEngine::ICNNNetwork& net)
         }
         plugin = InferenceEngine::InferencePlugin(enginePtr);
 
+        InferenceEngine::ICNNNetwork::InputShapes inShapes = netOwner.getInputShapes();
+        InferenceEngine::ICNNNetwork::InputShapes::iterator itr;
+        for (itr = inShapes.begin(); itr != inShapes.end(); ++itr)
+        {
+            if(inpBlobs.find(itr->first) == inpBlobs.end())
+            {
+                CV_Error(Error::StsAssert, "BlobMap key not exist in Blob");
+            }
+            itr->second[2] = inpBlobs[itr->first]->dims()[1];
+            itr->second[3] = inpBlobs[itr->first]->dims()[0];
+        }
+        netOwner.reshape(inShapes);
```

Comment:
I think we can also check at first if we really need to call this procedure. I mean if network's shape are matched with input ones there is no need to use it.